src/lib/dnssd/platform/Dnssd.h

```diff
@@ -209,6 +214,27 @@ CHIP_ERROR ChipDnssdFinalizeServiceUpdate();
 CHIP_ERROR ChipDnssdBrowse(const char * type, DnssdServiceProtocol protocol, chip::Inet::IPAddressType addressType,
                            chip::Inet::InterfaceId interface, DnssdBrowseCallback callback, void * context);
 
+/**
+ * This function stops service browsing.
+ *
+ * The type, protocol, addressType and interface parameters must match the ones used
+ * in the call to the `ChipDnssdBrowse` function. Alternatively, one shall be able
+ * to use "wildcard" values for the type (nullptr), protocol (kDnssdProtocolUnknown),
+ * addressType (IPAddressType::kAny) and interface (InterfaceId::Null()) to discard
+ * matches for particular browsing parameters.
+ *
+ * @param[in] type        The service type.
+ * @param[in] protocol    The service protocol.
+ * @param[in] addressType The protocol version of the IP address.
+ * @param[in] interface   The interface on which queries are sent.
+ *
+ * @retval CHIP_NO_ERROR                The stop browse succeeds.
+ * @retval Error code                   The stop browse fails.
+ *
+ */
+CHIP_ERROR ChipDnssdStopBrowse(const char * type, DnssdServiceProtocol protocol, chip::Inet::IPAddressType addressType,
+                               chip::Inet::InterfaceId interface);
+
```

Comment:
A method to stop browsing looks nice to me. That said in the current shape I'm sort of afraid that we it is easy to leak a `ResolverProxy`. Looking into the Tizen implementation it seems like the code does remove the matching contexts from the internal list of context but nobody will ever call `proxy->release()` on them.

Should we call the default callback for those contexts, passing a service size of `0` and with `CHIP_ERROR_END_OF_INPUT` ?

If no browse operation is in progress I assume that `CHIP_NO_ERROR` is returned too ? Can we add that to the comment ?

Reply:
>  Looking into the Tizen implementation it seems like the code does remove the matching contexts from the internal list of context but nobody will ever call proxy->release() on them.

Yes, that might be true, I will have to rethink that.

Also, for completeness, I though about adding browsing timeout to `ChipDnssdBrowse()`. I think it could be nice to be able to setup browsing for let say 15 seconds, then automatically stop it without the need of calling `ChipDnssdStopBrowse()`. Internally platform could implement it with a timer and call to `ChipDnssdStopBrowse`. What do you think about such addition?

Reply:
> Also, for completeness, I though about adding browsing timeout to ChipDnssdBrowse(). I think it could be nice to be able to setup browsing after let say 15 seconds, then automatically stop it without the need of calling ChipDnssdStopBrowse(). Internally platform could implement it with a timer and call to ChipDnssdStopBrowse. What do you think about such addition?

That sounds like a followup. I would have argued that sometimes someone may want to do long lived browse queries and may not desire this behaviour so it should be configurable at the API level.
My understanding is that most platforms at the moment are sort of terminating the browse instructions early, but it is unclear to me that the needs for long lived browse requests may not appears one day.

Reply:
>  I would have argued that sometimes someone may want to do long lived browse queries and may not desire this behaviour so it should be configurable at the API level.

Yes, that what I had in mind. Something like this (with proper type of course):
```
CHIP_ERROR ChipDnssdBrowse(const char * type, DnssdServiceProtocol protocol, chip::Inet::IPAddressType addressType,
                           chip::Inet::InterfaceId interface, DnssdBrowseCallback callback, void * context, duration timeout = kDurationInfinite);
```

Reply:
> Looking into the Tizen implementation it seems like the code does remove the matching contexts from the internal list of context but nobody will ever call proxy->release() on them.
> Should we call the default callback for those contexts, passing a service size of 0 and with CHIP_ERROR_END_OF_INPUT ?

For Tizen impl I've solved this problem exactly like that. Also, I've added a note to the `ChipDnssdStopBrowse` docstring:

```
Please note, that when stopping mDNS browsing, it is REQUIRED to call the callback
passed to the `ChipDnssdBrowse` function with an error code or the CHIP_END_OF_INPUT
code. This will allow the mDNS module to release its resources.
```

Reply:
I've dropped this new ChipDnssdStopBrowse API in favor of https://github.com/project-chip/connectedhomeip/pull/21553